lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -781,24 +781,48 @@ SILCombiner::visitInjectEnumAddrInst(InjectEnumAddrInst *IEAI) {
 
   // Ok, we have a payload enum, make sure that we have a store previous to
   // us...
-  SILBasicBlock::iterator II = IEAI->getIterator();
+  auto allocStackOperand = IEAI->getOperand();
   StoreInst *SI = nullptr;
   InitEnumDataAddrInst *DataAddrInst = nullptr;
   ApplyInst *AI = nullptr;
   Operand *EnumInitOperand = nullptr;
-  for (;;) {
-    if (II == IEAI->getParent()->begin())
+  for (auto usersIt = allocStackOperand->use_begin(); usersIt != allocStackOperand->use_end(); ++usersIt) {
```

Comment:
This violates LLVM coding standards:

http://llvm.org/docs/CodingStandards.html#turn-predicate-loops-into-predicate-functions

I would suggest rather than making an external function using a c++ higher kinded function.